### PR TITLE
Octeon: Update to 5.4 and add generate writable bootable images for ERL

### PR DESCRIPTION
**Pull request by diizzyy**
Adds 5.4 kernel config for octeon, sets 5.4 as default kernel version and generate bootlable USB images for EdgeRouter Lite

### CODE REVIEW

**Comment by adrianschmutzler** [comment]
Hi, it is preferred to do the target bump in a few steps, as this is much easier to review:
1. Just copy config and patches from 4.19 to 5.4 without changing them otherwise
2. Refresh 5.4 patches, so the adjustments due to the refresh become immediately obvious
3. Refresh 5.4 config, so ...
4. Bump kernel version for target (and maybe justify why no testing kernel is defined first)

File: target/linux/octeon/image/edgerouter_gen_usb_img.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
```
Comment (adrianschmutzler) [inline target/linux/octeon/image/edgerouter_gen_usb_img.sh:1]: Do we really need bash here?

File: target/linux/octeon/image/edgerouter_gen_usb_img.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+OUTPUT="$1"
+BOOTFS="$2"
+ROOTFS="$3"
+BOOTFSSIZE="$4"
+ROOTFSSIZE="$5"
+
+head=4
+sect=63
+
+set `ptgen -o $OUTPUT -h $head -s $sect -l 1024 -t b -p ${BOOTFSSIZE}M -t 83 -p ${ROOTFSSIZE}M`
```
Comment (adrianschmutzler) [inline target/linux/octeon/image/edgerouter_gen_usb_img.sh:18]: The backticks should be replaced by preferred $() syntax.

**Comment by diizzyy** [comment]
I have no idea what and why you're trying to enforce
https://github.com/openwrt/openwrt/commits/master/target/linux/archs38/config-5.4
https://github.com/openwrt/openwrt/commits/master/target/linux/mxs/config-5.4
https://github.com/openwrt/openwrt/commits/master/target/linux/octeontx/config-5.4
...and the list goes on

File: target/linux/octeon/image/edgerouter_gen_usb_img.sh (hunk 1 above)
Comment (diizzyy) [reply target/linux/octeon/image/edgerouter_gen_usb_img.sh]: Refreshed script to match ```/linux/sunxi/image/gen_sunxi_sdcard_img.sh```

File: target/linux/octeon/image/edgerouter_gen_usb_img.sh (hunk 2 above)
Comment (diizzyy) [reply target/linux/octeon/image/edgerouter_gen_usb_img.sh]: Refreshed script to match ```/linux/sunxi/image/gen_sunxi_sdcard_img.sh```

**Comment by diizzyy** [comment]
Compile/Runtime tested: octeon, EdgeRouter Lite, master

**Comment by adrianschmutzler** [comment]
Just look at how it was done for sunxi or kirkwood:
https://github.com/openwrt/openwrt/commits/master/target/linux/kirkwood
https://github.com/openwrt/openwrt/commits/master/target/linux/sunxi

> I have no idea what you're trying to enforce or why

For the why just refer to the reason I gave in my initial comment.

---

Despite that: Please think about your tone and attitude. I have given detailed instructions and reasons what to do. So either you try to argument why you do not want to follow these in a more substantial way, so one can have a real discussion, or just follow them, please.
Based on our previous encounters I have the strong suspicion that you just don't want to understand my feedback, as this implies that you would to have to change something. I won't go on explaining stuff to you three or four times just to make you accept that I really want that done.

**Comment by adrianschmutzler** [comment]
BTW: Shouldn't the folder by patches-5.4 instead of patch-5.4?

**Comment by diizzyy** [comment]
While your reviews certainly can be useful you also include a lot of your own preferences which aren't applied by other members with commit access and honestly is getting very tiresome at this point especially since it doesn't really go anywhere/matters in the end.

**Comment by adrianschmutzler** [comment]
> which aren't applied by other members

Note that splitting target bumps like I suggested has also been requested by others committers for several other 5.4 PRs. I admit that it's not really necessary for the patches here, but it could be helpful for reviewing the config.

>  honestly is getting very tiresome at this point especially since it doesn't really go anywhere/matters in the end.

That depends on what your goals are (i.e. what "matters"). If your goal is just to get stuff in without caring about quality or guidance for other people looking at it in the future, then your way might indeed be the most efficient one: Just do what's absolutely necessary and then wait for the committer with the lowest standards.
However, my aim is to not just merge anything "that works", but to also provide a certain quality and tidyness of code and history. I've learned that while this might seem labor-intensive and annoying in the first place, eventually it will mean less work for all people together. Thus, I think it's an act of care and courtesy towards reviewers and future readers of the patches.

I admit that this particular PR might be not the perfect subject for that discussion, as in this case I was merely suggesting an improvement of your code to help reviewers (as there are indeed different practices here; I was mostly annoyed because I provided clear instructions and you just told you don't know what to do). The discussion would actually belong into #2754 , where you put up the same fight over an actually codified requirement just because you didn't want to write three sentences.

I will continue to request certain adjustments to enhance quality based on my view. If you don't want to follow them because you hope for another committer with lower standards, please say that right away next time, so I do not waste my time on trying to convince you where you did not intend to argue in the first place.

**Comment by adrianschmutzler** [comment]
To be fair, I admit that my use of "it is preferred" will not sustain a check by scientific measures.

Anyway, to not get the important stuff hidden by that clash, please fix the patches-5.4 directory and re-test if possible, as I suspect that the patches haven't been applied when located in "patch-5.4"?

**Comment by diizzyy** [comment]
I do understand your intentions however the continuous remarks without any written guidelines makes it very frustrating and time consuming especially when you see "drive-by" commits. Reviews are good and I appreciate your time however commit messages tends to go into nitpicking or almost just for the sake of it in comparison.

I've already fixed the directory and it works fine on my end.

**Comment by diizzyy** [comment]
@hauke 
Can you have a look at this? It would be nice to not have this PR bitrot like the previous one adding support for writeable images.

**Comment by diizzyy** [comment]
..and for the sake if it, @ynezz

**Comment by adrianschmutzler** [comment]
@diizzyy Since you have not even updated your obviously wrong patch folder name while requesting additional reviews, I've taken matters in my own hands and pushed an updated branch to my staging area: https://git.openwrt.org/?p=openwrt/staging/adrian.git;a=shortlog;h=refs/heads/octeon

I will build-test it soon. However, I still need some information:
- Above, you stated that you updated patch-5.4 to patches-5.4 for testing. Did you really do this and just forgot to push? If not, please re-test.
- Please mention which device(s) you tested on, so I can add this to the commit description of the kernel bump.

Note that I didn't and won't review/include you second commit.

**Comment by diizzyy** [comment]
@adrianschmutzler 
Really?
https://github.com/openwrt/openwrt/pull/2900/files#diff-1ed799721d94093dc314d9b3d0785b61
https://github.com/openwrt/openwrt/pull/2900/files#diff-6a01f2083400fad2a8f5945e4c472052
https://github.com/openwrt/openwrt/pull/2900/files#diff-3dc7d710e91846b6c6b051d37c19c950

https://patch-diff.githubusercontent.com/raw/openwrt/openwrt/pull/2900.patch
```
 rename target/linux/octeon/{patch-5.4 => patches-5.4}/100-ubnt_edgerouter2_support.patch (100%)
 rename target/linux/octeon/{patch-5.4 => patches-5.4}/110-er200-ethernet_probe_order.patch (100%)
 rename target/linux/octeon/{patch-5.4 => patches-5.4}/120-cmdline-hack.patch (100%)
````

As for testing:
https://github.com/openwrt/openwrt/pull/2900#issuecomment-609717768
https://github.com/openwrt/openwrt/pull/2900#issuecomment-609847597

**Comment by adrianschmutzler** [comment]
Well, I didn't look for the rename in the wrong patch; it's actually in commit 2 where it doesn't belong at all.

So, after having walking through your forest of links, I assume the answer to my questions would have been the following:

> Above, you stated that you updated patch-5.4 to patches-5.4 for testing. Did you really do this and just forgot to push? If not, please re-test.

"Yes, I tested it with the changed path, but only applied it to the second commit in this PR."

> Please mention which device(s) you tested on, so I can add this to the commit description of the kernel bump.

"I run-tested this on the EdgeRouter Lite."

If these are not the right answers, feel free to correct me.
Anyway, there won't be anything happening fast as my dev server's power supply broke today and I will receive a replacement earliest on Tuesday, where I will start with looking what's left of the MB and drives then.

**Comment by diizzyy** [comment]
Feel free to do whatever you want, I'm done spending time on your requests if you can't even bother to read.

**Comment by diizzyy** [comment]
While at it, let's close it and make someone else (if ever) redo the work.

**Comment by oldium** [comment]
@diizzyy You are acting like a child, but I would not be surprised if you have some sort of mental illness. But that is fine, lot of IT people have some problems in social life, but are good in programming. You just need to find a way how to learn saying "OK, I will do it" instead of "fuck-off, I will close it". Especially when you are contributing to repository of someone else - their repository, their rules. It is not like "hurray, a contribution", but "check it, make it readable to others, accept". This is not a child project.

**Comment by diizzyy** [comment]
@oldium 
Going for personal accusations especially medical ones would most likely mean you're violating https://openwrt.org/rules (#12) and by other projects standard much more severe. :-) I fail to see the relevance of your opinion in this matter and doesn't really seem to make a point. If someone won't even bother to read perhaps you should do something else, don't you think? I'm sure what rules you're referring to? That said, even if I did fix the commit I have very high doubts it would make it in the tree anyway so I've decided withdraw my contribution instead of letting it bitrot which I would like to think that I'm entitled to do.

**Comment by oldium** [comment]
@dizzyy The commits in a Pull Request should not be a random set of changes, where the important part is the final state, but it should be a step-by-step work towards the end. You made a mistake in a first commit and fixed it in the second commit. Having the error and the fix part of a Pull Request is wrong and makes reviewing hard. This point was proven in this discussion. I recommend that you learn how to do Git Rebase, so that the commits contain only work, which is “correct” without the need to fix anything later. Next time please use git rebase to rework your first commit and do a force push.

When you say that you have “doubts it would make it in the tree” you again failed to see that @adrianschmutzler wanted to help you, took your changes into his tree and tried to work on it (compile/test). It did not happen just because there were some hadrware problems, but that did not stop Adrian from explaining it to you. That was also an excuse why it will take a little longer.

Other than that I just said that having any kind of mental illness is completely fine and being childish is not good. And you see that I am still very polite in the discussion, saying you what you did wrong and what you should change in order to have your next Pull Request accepted faster.

**Comment by oldium** [comment]
Just please focus more on the things you are asked to do (and do them) and less on finding excuses.

**Comment by diizzyy** [comment]
@oldium 
Please just stop trying to justify your actions and we'll leave it that as you're not contributing in any way.

**Comment by oldium** [comment]
I said what you did wrong. Please start doing what others are telling you to do.

**Comment by adrianschmutzler** [comment]
For the record: When my machine is up again, I will merge my referenced tree with testing kernel support. Let's see where it goes from there.
https://git.openwrt.org/?p=openwrt/staging/adrian.git;a=shortlog;h=refs/heads/octeon

**Pull request closed without merging**